core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -896,33 +903,27 @@ private Iterable<ManifestFile> prepareNewManifests() {
         manifest -> GenericManifestFile.copyOf(manifest).withSnapshotId(snapshotId()).build());
   }
 
-  private ManifestFile newFilesAsManifest() {
-    if (hasNewFiles && cachedNewManifest != null) {
-      deleteFile(cachedNewManifest.path());
-      cachedNewManifest = null;
+  private List<ManifestFile> newFilesAsManifest() {
```

Comment:
This should also fix method naming. If a method returns more than one item, it should be plural.

Reply:
Updated